File: chat-uikit/src/main/java/io/agora/chat/uikit/chatthread/widget/EaseChatRowThreadNotify.java
package io.agora.chat.uikit.chatthread.widget;

import android.content.Context;
import android.text.Spannable;
import android.text.SpannableStringBuilder;
import android.text.style.ForegroundColorSpan;
import android.widget.BaseAdapter;
import android.widget.TextView;


import io.agora.chat.ChatMessage;
import io.agora.chat.TextMessageBody;
import io.agora.chat.uikit.R;
import io.agora.chat.uikit.widget.chatrow.EaseChatRowText;

/**
 * big emoji icons
 *
 */
public class EaseChatRowThreadNotify extends EaseChatRowText {
    private TextView tv_thread_notify;

    public EaseChatRowThreadNotify(Context context, boolean isSender) {
        super(context, isSender);
    }

    public EaseChatRowThreadNotify(Context context, ChatMessage message, int position, BaseAdapter adapter) {
        super(context, message, position, adapter);
    }

    @Override
    protected void onInflateView() {
        inflater.inflate(R.layout.ease_layout_row_thread_notify, this);
    }

    @Override
    protected void onFindViewById() {
        tv_thread_notify = (TextView) findViewById(R.id.tv_thread_notify);
    }

    @Override
    public void onSetUpView() {
        if(tv_thread_notify == null) {
            return;
        }
        TextMessageBody textBody = (TextMessageBody) message.getBody();
        String message = textBody.getMessage();
        SpannableStringBuilder builder = new SpannableStringBuilder(message);
        int length = context.getString(R.string.ease_join_the_thread).length();
        builder.setSpan(new ForegroundColorSpan(this.getResources().getColor(R.color.ease_color_brand)), message.length() - length, message.length(),
                Spannable.SPAN_EXCLUSIVE_EXCLUSIVE);
        tv_thread_notify.setText(builder);
    }
}

File: ease-im-kit/src/main/kotlin/io/agora/uikit/widget/chatrow/EaseChatRowThreadNotify.kt
package io.agora.uikit.widget.chatrow

import android.content.Context
import android.text.SpannableString
import android.text.TextPaint
import android.text.method.LinkMovementMethod
import android.text.style.ClickableSpan
import android.util.AttributeSet
import android.view.View
import android.widget.TextView
import io.agora.uikit.EaseIM
import io.agora.uikit.R
import io.agora.uikit.common.ChatClient
import io.agora.uikit.common.EaseConstant
import io.agora.uikit.feature.thread.EaseChatThreadActivity
import io.agora.uikit.model.EaseProfile

class EaseChatRowThreadNotify @JvmOverloads constructor(
    private val context: Context,
    private val attrs: AttributeSet? = null,
    private val defStyleAttr: Int = 0,
    isSender: Boolean = false
): EaseChatRow(context, attrs, defStyleAttr, isSender)  {
    private val contentView: TextView? by lazy { findViewById(R.id.text_content) }

    override fun onInflateView() {
        inflater.inflate(R.layout.ease_row_thread_notify, this)
    }

    override fun onSetUpView() {
        message?.run {
            if (ext().containsKey(EaseConstant.THREAD_TOPIC_MESSAGE_ID)){
                var creator:String? = from
                var content = ""
                EaseProfile.getGroupMember(conversationId(),from)?.let { profile ->
                    creator = profile.getRemarkOrName()
                }
                val coloredText = context.getString(R.string.ease_thread_notify_detail)
                val topicMsgId = getStringAttribute(EaseConstant.THREAD_TOPIC_MESSAGE_ID)
                val topicMsg = ChatClient.getInstance().chatManager().getMessage(topicMsgId)
                topicMsg?.let {
                    val threadName = it.chatThread?.chatThreadName?:""
                    content = "${context.getString(R.string.ease_thread_notify_content,creator,threadName)} $coloredText"
                    val spannableString = SpannableString(content)
                    val startIndex = content.indexOf(coloredText)
                    val endIndex = startIndex + coloredText.length
                    val clickableSpan = object : ClickableSpan() {
                        override fun onClick(view: View) {
                            EaseIM.getConfig()?.chatConfig?.enableChatThreadMessage?.let {
                                if (!it) {
                                    return
                                }
                            }
                            it.chatThread?.let { thread->
                                EaseChatThreadActivity.actionStart(
                                    context = context,
                                    conversationId = thread.parentId,
                                    threadId = thread.chatThreadId,
                                    topicMsgId = topicMsgId,
                                )
                            }
                        }

                        override fun updateDrawState(ds: TextPaint) {
                            ds.color = resources.getColor(R.color.ease_color_primary)
                            ds.isUnderlineText = false
                        }
                    }
                    spannableString.setSpan(clickableSpan, startIndex, endIndex, 0)
                    contentView?.movementMethod = LinkMovementMethod.getInstance()
                    contentView?.text = spannableString
                }
            }
        }
    }
}